File: python/utils/validateAndCheckType.py
import os
import numpy as np
from scipy.io import loadmat

def validate_and_check_type(file, field, type_check):
    """
    Load a variable from a .mat file and validate its type.
    
    Parameters
    ----------
    file : str
        Path to a .mat file.
    field : str
        Name of the variable to extract from the file.
    type_check : str
        Expected type of the variable. Supported values:
        'numeric' : checks that the variable is numeric.
        'float'   : checks that the variable is floating-point.
    
    Returns
    -------
    variable : array-like
        The extracted variable if all validations pass.
    
    Raises
    ------
    FileNotFoundError
        If the file does not exist.
    KeyError
        If the specified field is not found in the MAT-file.
    TypeError
        If the variable does not match the requested type.
    """
    # Check if file exists
    if not os.path.isfile(file):
        raise FileNotFoundError(f'File not found: {file}')
    
    # Load the .mat file
    variable = loadmat(file)
    
    # Check if field exists
    if field not in variable:
        raise KeyError(f'Field {field} does not exist in the loaded variable.')
    
    variable = variable[field]
    
    # Check type of variable
    if type_check == 'numeric':
        if not np.issubdtype(variable.dtype, np.number):
            raise TypeError(f'Variable is not numeric: {field}')
    elif type_check == 'float':
        if not np.issubdtype(variable.dtype, np.floating):
            raise TypeError(f'Variable is not a float: {field}')
    
    return variable

File: matlab/utils/validateAndCheckType.m
function variable = validateAndCheckType( file, field, type_check )
%VALIDATEANDCHECKTYPE Load a variable from a .mat file and validate its type.
%
%   variable = VALIDATEANDCHECKTYPE(file, field, type_check) loads the specified
%   MAT-file, extracts the variable named by 'field', and verifies that it
%   matches the expected data type.
%
%   Inputs:
%       file  - Path to a .mat file.
%       field - Name of the variable to extract from the file.
%       type_check  - Expected type of the variable. Supported values:
%               'numeric' : checks that the variable is numeric.
%               'float'   : checks that the variable is floating-point.
%
%   Output:
%       variable - The extracted variable if all validations pass.
%
%   The function throws an error if:
%       - The file does not exist.
%       - The specified field is not found in the MAT-file.
%       - The variable does not match the requested type.

assert(isfile(file), 'File not found: %s', file);
variable = load(file);

% Check and load field
if ~isfield(variable, field)
    error('Field %s does not exist in the loaded variable.', field);
end
variable = variable.(field);

% Check type of variable
if strcmp(type_check, 'numeric')
    assert(isnumeric(variable), 'Variable is not a numeric: %s', field);
end
if strcmp(type_check, 'float')
    assert(isfloat(variable), 'Variable is not a float: %s', field);
end
